lab-sharmarke/server.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const http = require('http');
+const url = require('url');
+const querystring = require('querystring');
+const cowsay = require('cowsay');
+const parseBody = require('./lib/parse-body.js');
+const PORT = process.env.PORT || 3000;
+
+const server = http.createServer(function(req, res) {
+  req.url = url.parse(req.url);
+  req.url.query = querystring.parse(req.url.query);
+
+  if (req.method === 'POST' && req.url.pathname === '/') {
+    parseBody(req, function(err) {
+      if (err) {
+        res.writeHead(400);
+        res.write(cowsay.say({text: 'POST request bad'}));
+        res.end();
+      } else {
+        res.writeHead(200, {
+          'Content-Type:': 'text/plain'
+        });
+        res.end();
+      }
+    });
+  }
+
+  if(req.method === 'GET' && req.url.pathname ==='/cowsay') {
+    let params = req.url.query;
+    if(params.text) {
+      res.writeHead(200, {
+        'Content-Type': 'text/plain'
+      });
+      res.write(cowsay.say({text: params.query}));
+      res.end();
+    } else {
+      res.writeHead(400, {
+        'Content-Type': 'text/plain'
+      });
+      res.write(cowsay.say({text: 'GET request is bad'}));
+      res.end();
+    }
+  }
+
+  if (req.url.pathname ==='/') {
```

Comment:
with this as the same pathname as the post if you make a post to / this will sometimes hit causing a 'write after end error' when you close the response a 2nd time.